### PR TITLE
Update: add fixer for `wrap-iife`

### PR DESCRIPTION
**Pull request by not-an-aardvark**
**What is the purpose of this pull request? (put an "X" next to item)**

[ ] Documentation update
[ ] Bug fix ([template](https://github.com/eslint/eslint/blob/master/templates/bug-report.md))
[ ] New rule ([template](https://github.com/eslint/eslint/blob/master/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://github.com/eslint/eslint/blob/master/templates/rule-change-proposal.md))
[x] Add autofixing to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

**Please check each item to ensure your pull request is ready:**

- [x] I've read the [pull request guide](http://eslint.org/docs/developer-guide/contributing/pull-requests)
- [x] I've included tests for my change
- [x] I've updated documentation for my change (if appropriate)

**What changes did you make? (Give an overview)**

This adds a fixer for [`wrap-iife`](http://eslint.org/docs/rules/wrap-iife).

If the IIFE has no surrounding parentheses, the fixer surrounds it with parentheses. The fixer also removes parentheses that are in the wrong place according to the rule option:

```js
/* eslint wrap-iife: [2, "outside"] */

var foo = function() {}();
(function() {})();

// gets fixed to:

var foo = (function() {}());
(function() {}());
```

```js
/* eslint wrap-iife: [2, "inside"] */

var foo = function() {}();
(function() {}());

// gets fixed to:

var foo = (function() {})();
(function() {})();
```

**Is there anything you'd like reviewers to focus on?**

Nothing in particular.

### CODE REVIEW

File: lib/rules/wrap-iife.js [hunk 1]
```diff
@@ -52,11 +54,51 @@ module.exports = {
...
-                        context.report(node, "Wrap only the function expression in parens.");
+                        context.report({
+                            node,
+                            message: "Wrap only the function expression in parens.",
+                            fix(fixer) {
+
+                                // The outer call expression will always be wrapped at this point.
+                                // Replace the range between the end of the function expression and the end of the call expression.
+                                // for example, in `(function(foo) {}(bar))`, the range `(bar))` should get replaced with `)(bar)`.
+                                // Replace the parens from the outer expression, and parenthesize the function expression.
```
Comment (platinumazure) [inline lib/rules/wrap-iife.js]: In our [coding conventions](http://eslint.org/docs/developer-guide/code-conventions), we prefer multi-line comments to be formatted like this:

```js
/*
 * First line of comment here.
 * Second line. Note that each line starts with asterisk aligned with the open-comment asterisk character.
 * Both comment start and end should be on their own line.
 * Comment ends with asterisk of close-comment token aligned with other asterisks.
 */
```

File: lib/rules/wrap-iife.js [hunk 2]
```diff
@@ -52,11 +54,51 @@ module.exports = {
...
-                        context.report(node, "Move the invocation into the parens that contain the function.");
+                        context.report({
+                            node,
+                            message: "Move the invocation into the parens that contain the function.",
+                            fix(fixer) {
+
+                                // The inner function expression will always be wrapped at this point.
+                                // It's only necessary to replace the range between the end of the function expression
+                                // and the call expression. For example, in `(function(foo) {})(bar)`, the range `)(bar)`
+                                // should get replaced with `(bar))`.
```
Comment (platinumazure) [inline lib/rules/wrap-iife.js]: Same as above re: multiline comment.

**Comment by not-an-aardvark** [comment]
Added a commit to fix multiline comment style.

**Comment by platinumazure** [comment]
Thanks @not-an-aardvark, changes LGTM.

**Comment by kaicataldo** [comment]
I'll champion this.

**Comment by kaicataldo** [comment]
LGTM. Thanks for contributing to ESLint!

**Pull request merged**